src/ros_can_nodes/src/RosCanNode.cpp

```diff
@@ -278,49 +278,29 @@ namespace roscan {
 
         canid_t header = 0x0;
 
+        printf("can topic id 0x%x\n", topicID);
         header |= (1 << ROSCANConstants::Common::bitshift_mode);
         header |= (0 << ROSCANConstants::Common::bitshift_priority);
         header |= (0 << ROSCANConstants::Common::bitshift_func);
+        header |= (((uint32_t)topicID) << ROSCANConstants::ROSTopic::bitshift_topic_id);
         header |= (0 << ROSCANConstants::Common::bitshift_seq);
-        if(id_ == 1){
-	        header |= ((topicID * 2) << ROSCANConstants::ROSTopic::bitshift_topic_id);
-        } else if (id_ == 0 ){
-            header |= ((topicID * 2 + 1) << ROSCANConstants::ROSTopic::bitshift_topic_id);
-	    } else if (id_ == 4) {
-            header |= ((topicID + 8) << ROSCANConstants::ROSTopic::bitshift_topic_id);
-        } else {
-            header |= (topicID << ROSCANConstants::ROSTopic::bitshift_topic_id);
-        }
+        printf("topic id at position 0x%x\n", (((uint32_t)topicID) << ROSCANConstants::ROSTopic::bitshift_topic_id));
```

Comment:
@YiweiHan the changes to headers here is relevant for you work as well. The ones on master had a few problems

Reply:
oh cool, this function is much cleaner. i didn't have a clue what was going on before anyway